release_tools/changelog.py

```diff
@@ -0,0 +1,329 @@
+
+    def __init__(self, title, category, author, pr=None, notes=None):
+        self.title = title
+        self.category = category
+        self.author = author
+        self.pr = pr
+        self.notes = notes
+
+    def to_dict(self):
+        return {
+            'title': self.title,
+            'category': self.category,
+            'author': self.author,
+            'pull_request': self.pr,
+            'notes': self.notes
+        }
+
+
+def title_prompt():
+    prompt_msg = ">> Please specify the title of your change"
+    return prompt_msg
+
+
+def category_prompt():
+    prompt_msg = ">> Please specify the category of your change\n\n"
+    prompt_msg += "\n".join(["{}. {}".format(cat.value, cat.title) for cat in CategoryChange])
+    prompt_msg += "\n\n"
+    return prompt_msg
+
+
+def validate_title(ctx, param, value):
+    """Check title option values."""
+
+    value = value.strip("\n\r ")
+
+    if not value:
+        raise click.BadParameter("title cannot be empty")
+    return value
+
+
+def validate_category(ctx, param, value):
+    """Check category option values.
+
+    Valid values for a category are integer indexes
+    and strings.
+    """
+    # Check if the value is an index
+    try:
+        value = int(value)
+    except ValueError:
+        # The value is not an index.
+        # Check if it is a valid category string.
+        try:
+            CategoryChange[value.upper()]
+        except KeyError:
+            msg = "valid options are {}".format(CategoryChange.values())
+            raise click.BadParameter(msg)
+        else:
+            return value
+
+    # Check if the index is in range.
+    try:
+        category = CategoryChange(value).category
+        return category
+    except ValueError:
+        msg = "please select an index between 1 and {}".format(len(CategoryChange))
+        raise click.BadParameter(msg)
+
+
+@click.command()
+@click.option('-t', '--title', prompt=title_prompt(),
+              callback=validate_title,
+              help="Title for the changelog entry.")
+@click.option('-c', '--category', prompt=category_prompt(),
+              callback=validate_category,
+              help="The category of the change.")
+@click.option('--dry-run', is_flag=True,
+              help="Do not generate an entry. Print to the standard output instead.")
+@click.option('--overwrite', is_flag=True,
+              help="Force to replace an existing entry.")
+@click.option('--editor/--no-editor', default=True,
+              help="Open entry in the default editor.")
+def changelog(title, category, dry_run, overwrite, editor):
+    """Interactive tool to create unreleased Changelog entries.
+
+    This tool will help you to create valid Changelog entries
+    for a Git repository. You will need to run this script inside
+    that repository.
+
+    It will guide you to create a new entry. At the end of the process,
+    a text editor will open to let you review the entry and make the
```

Comment:
It would be helpful to add some info about the editor (how to insert, save and quit)

Reply:
The editor is the default editor configured in your system. I added a comment regarding that in the docstring.